daemon/execdriver/native/driver.go

```diff
@@ -206,10 +204,12 @@ func waitInPIDHost(p *libcontainer.Process, c libcontainer.Container) func() (*o
 			}
 			s = execErr.ProcessState
 		}
+
+		processes, err := c.Processes()
```

Comment:
Shouldn't you be freezing the container, getting the pids, sending the signals, then unfreeze and wait on the individual processes like the code does in container.Destroy()?